hpx/runtime/threads/executors/current_executor.hpp

```diff
@@ -21,11 +21,11 @@ namespace hpx { namespace threads { namespace executors
     namespace detail
     {
         //////////////////////////////////////////////////////////////////////
-        class HPX_EXPORT generic_thread_pool_executor
+        class HPX_EXPORT current_executor
           : public threads::detail::scheduled_executor_base
         {
         public:
-            generic_thread_pool_executor(policies::scheduler_base* scheduler);
+            current_executor(policies::scheduler_base* scheduler);
```

Comment:
Why do we have a `current_executor` and a `this_thread_executor` now, where is the difference?